### PR TITLE
Move remainder of Roslyn to new SDK

### PR DESCRIPTION
**Pull request by jaredpar**
This change will move the remainder of Roslyn to the new SDK.  

Reviewers are encouraged to look at the individual commits here. The majority of the change is an automated transformation using the ConvertPackageRef tool. That is isolated to a single commit. 

https://github.com/jaredpar/ConvertPackageRef

### CODE REVIEW

File: build/scripts/edit-designers.ps1 [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# The new project system does not support designers. This blocks our ability to make XAML
+# changes through the designer. A rare event but necessary on occasiona.
```
Comment (tmat) [inline build/scripts/edit-designers.ps1]: >a [](start = 70, length = 1)

typo

Comment (jaredpar) [inline build/scripts/edit-designers.ps1]: Oh .. I'm sure you'll find more of those 😄

**Comment by jaredpar** [comment]
retest windows_build_correctness_prtest please

**Comment by jaredpar** [comment]
retest windows_build_correctness_prtest please

File: build/Targets/Settings.props [hunk 2]
```diff
@@ -49,6 +49,11 @@
     <VisualStudioCodename>Dev$(VisualStudioReferenceMajorVersion)</VisualStudioCodename>
     <MinimumVisualStudioVersion>$(VisualStudioVersion)</MinimumVisualStudioVersion>
 
+    <!-- When building WPF / VSIX projects MSBuild will create a temporary project with an extension of
+         tmp_proj.  In that case the SDK is unable to determine the target language and cannot pick
+         the correct import.  Need to set it explicitly here. -->
+    <LanguageTargets Condition="'$(ProjectLanguage)' != ''">$(MSBuildToolsPath)\Microsoft.$(ProjectLanguage).targets</LanguageTargets>
```
Comment (tmat) [inline build/Targets/Settings.props]: >ProjectLanguage [](start = 35, length = 15)

Instead of setting a property manually, can we check the extension of the first file in Compile items?

Comment (jaredpar) [inline build/Targets/Settings.props]: Technically compile items can be empty and extension doesn't necessarily mean language. Crazy people could put C# code into .vb files if they choose. This infrastructure was already in place hence just leveraging it for this change.

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure please review

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jaredpar** [comment]
The two retests were jenknis issues.

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jaredpar** [comment]
another jenkins issues

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jaredpar** [comment]
For anyone following along here, I'm not being lazy and just issuing retest without looking. I've just managed to hit five Jenkins infrastructure issues in a row. 😦

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

File: build/Targets/Imports.targets [hunk 3]
```diff
@@ -138,6 +138,10 @@
     <DisableImplicitFrameworkReferences Condition="'$(DisableImplicitFrameworkReferences)' == ''" >true</DisableImplicitFrameworkReferences>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
+      
+    <!-- Only generate our runtimeconfig.json files for net core apps. It's unnecessary in desktop projects
+         but gets included in lots of output items like VSIX. -->
+    <GenerateRuntimeConfigurationFiles Condition="'$(TargetFramework)' != 'netcoreapp1.1' AND '$(TargetFramework)' != 'netcoreapp2.0'">false</GenerateRuntimeConfigurationFiles>
```
Comment (tmat) [inline build/Targets/Imports.targets]: Seems like something that should be fixed in SDK. Is there an issue tracking it?

Comment (jaredpar) [inline build/Targets/Imports.targets]: It's fixed in the 2.0 SDK.

**Comment by jaredpar** [comment]
ping @dotnet/roslyn-infrastructure. Been 24 hours now. This is a very unweildy change that keeps hitting merge conflicts. Want to get this in.

File: src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj [hunk 4]
```diff
@@ -11,7 +11,7 @@
     <RootNamespace>Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests</RootNamespace>
     <AssemblyName>Roslyn.Compilers.CSharp.CommandLine.UnitTests</AssemblyName>
     <AllowUnsafeBlocks>false</AllowUnsafeBlocks>
-    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
+    <TargetFramework>net46</TargetFramework>
     <RuntimeIdentifiers>win7</RuntimeIdentifiers>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
```
Comment (sharwell) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: :bulb: `<ProjectTypeGuids>` should be removed from the project files with this change.

File: src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj [hunk 5]
```diff
@@ -11,7 +11,7 @@
...
     <RoslynProjectType>UnitTest</RoslynProjectType>
```
Comment (sharwell) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: :question: `UnitTest`/`UnitTestPortable` refers to the target framework, not the SDK, correct?

File: src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs [hunk 6]
```diff
@@ -21,7 +21,7 @@ public class WriteCodeFragmentEx : Task
 
         public override bool Execute()
```
Comment (sharwell) [inline src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs]: :question: Do we still need this task or is it available in the versions of MSBuild we're using now?

File: src/Tools/RepoUtil/RepoUtil.csproj [hunk 7]
```diff
@@ -10,7 +10,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>RepoUtil</RootNamespace>
     <AssemblyName>RepoUtil</AssemblyName>
-    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
+    <TargetFramework>net46</TargetFramework>
     <RuntimeIdentifiers>win7</RuntimeIdentifiers>
     <UseVSHostingProcess>false</UseVSHostingProcess>
     <TargetFrameworkProfile />
```
Comment (sharwell) [inline src/Tools/RepoUtil/RepoUtil.csproj]: :bulb: `<TargetFrameworkProfile>` can be removed now.

**Comment by sharwell** [comment]
I understand the desire to stop maintaining this branch separately from the main line. I do hope that we don't stop working on the performance of mixed solution scenarios though, as I believe they better reflect the configurations external users are more likely to be working with.

File: src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj (hunk 4 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: Can you elaborate on why? I'm unfamiliar with what these do and how CPS makes it okay to remove them.

File: src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj (hunk 5 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: Correct. Refers to what is being targeted: desktop or portable. This still targets desktop.

File: src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj (hunk 4 above)
Comment (sharwell) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: CPS doesn't use them. CPS uses `<ProjectCapability>` instead. Anything that is looked at `<ProjectTypeGuids>` will not be correct in the result.

Comment (sharwell) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: @srivatsn Can you confirm my reply :arrow_up:

File: src/Tools/RepoUtil/RepoUtil.csproj (hunk 7 above)
Comment (jaredpar) [inline src/Tools/RepoUtil/RepoUtil.csproj]: Good catch. Will remove.

File: src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs (hunk 6 above)
Comment (jaredpar) [inline src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs]: This can be removed now. They fixed it in MSBuild 15.0

File: src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj (hunk 4 above)
Comment (srivatsn) [inline src/Compilers/CSharp/Test/CommandLine/CSharpCommandLineTest.csproj]: Ya ProjectTypeGuids are only used by the old PS. Not needed anymore.

**Comment by srivatsn** [comment]
> I do hope that we don't stop working on the performance of mixed solution scenarios though,

I agree. I suppose the dev15.3.x branch will always be mixed mode and should be easy to test as we make any changes to make that scenario better.

File: src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs (hunk 6 above)
Comment (sharwell) [inline src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs]: I assume you'll do this in a separate PR. Want me to file a bug?

File: src/Tools/RepoUtil/RepoUtil.csproj (hunk 7 above)
Comment (srivatsn) [inline src/Tools/RepoUtil/RepoUtil.csproj]: There are more properties that can go away either because they are defaulted or not used anymore. We can also convert to file globs and get rid of the individual file includes - (it is a big time saver for merges\switching branches). That should be a separate PR though. I'll volunteer to do that (will be good testing for my [tool](https://github.com/srivatsn/ProjectSimplifier)) as long as you don't object to globs.

File: src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs (hunk 6 above)
Comment (jaredpar) [inline src/Tools/BuildUtil/BuildUtil/WriteCodeFragmentEx.cs]: @sharwell filed the following to track the two big cleanup items. Will likely do before I leave today. 

https://github.com/dotnet/roslyn/issues/20526

File: src/Tools/RepoUtil/RepoUtil.csproj (hunk 7 above)
Comment (jaredpar) [inline src/Tools/RepoUtil/RepoUtil.csproj]: What other properties can go away? Would like to delete + ban them with BuildBoss. 

I don't have any objections to globs. It was on my "TODO" list once I got this change merged.

File: Compilers.sln [hunk 8]
```diff
@@ -1,185 +1,161 @@
...
+
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
 VisualStudioVersion = 15.0.26405.2
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CodeAnalysisTest", "src\Compilers\Core\CodeAnalysisTest\CodeAnalysisTest.csproj", "{A4C99B85-765C-4C65-9C2A-BB609AAB09E6}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CodeAnalysisTest", "src\Compilers\Core\CodeAnalysisTest\CodeAnalysisTest.csproj", "{A4C99B85-765C-4C65-9C2A-BB609AAB09E6}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CodeAnalysis", "src\Compilers\Core\Portable\CodeAnalysis.csproj", "{1EE8CAD3-55F9-4D91-96B2-084641DA9A6C}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CodeAnalysis", "src\Compilers\Core\Portable\CodeAnalysis.csproj", "{1EE8CAD3-55F9-4D91-96B2-084641DA9A6C}"
```
Comment (jasonmalinowski) [inline Compilers.sln]: Were these missed in the previous round of this?

File: build/Targets/Imports.targets (hunk 3 above)
Comment (jasonmalinowski) [inline build/Targets/Imports.targets]: Should we have a comment then that this can be removed when we move to 2.0?

File: build/Targets/Imports.targets [hunk 9]
```diff
@@ -245,6 +249,16 @@
     </SuggestedBindingRedirects>
   </ItemGroup>
 
+  <!-- 
+       Work around some UI delays in the new project system
+
+       https://github.com/dotnet/project-system/issues/2297
```
Comment (jasonmalinowski) [inline build/Targets/Imports.targets]: This bug is already closed. Can the comment be updated to state what criteria we're still waiting for on this one?

File: build/Targets/Imports.targets [hunk 10]
```diff
@@ -353,6 +369,7 @@
 
   <Import Project="GenerateAssemblyInfo.targets" Condition="'$(ProjectLanguage)' == 'CSharp' OR '$(ProjectLanguage)' == 'VB'" />
   <Import Project="GenerateInternalsVisibleTo.targets" />
+  <Import Project="DisableTransitiveReferences.targets" Condition="'$(RoslynProjectType)' == 'Vsix' AND '$(RoslynSdkProject)' == 'true'" />
```
Comment (jasonmalinowski) [inline build/Targets/Imports.targets]: Why the conditioning on Vsix projects here? I imagine much profanity in your direction once somebody else gets bit by this...

File: build/Targets/Settings.props [hunk 11]
```diff
@@ -167,6 +174,7 @@
         <RemoveIntegerChecks>true</RemoveIntegerChecks>
         <RootNamespace></RootNamespace>
         <MSBuildTargetsLanguageName>VisualBasic</MSBuildTargetsLanguageName>
+        <NoWarn>$(NoWarn);40057</NoWarn>
```
Comment (jasonmalinowski) [inline build/Targets/Settings.props]: I see this previously existed in some places, now we're applying it everywhere. Why? Can we add a comment for what this is working around?

File: build/scripts/build.ps1 [hunk 12]
```diff
@@ -207,10 +207,12 @@ function Test-XUnit() {
         Deploy-VsixViaTool
     }
 
+    $logFilePath = Join-Path $configDir "runtests.log"
     $unitDir = Join-Path $configDir "UnitTests"
     $runTests = Join-Path $configDir "Exes\RunTests\RunTests.exe"
     $xunitDir = Join-Path (Get-PackageDir "xunit.runner.console") "tools"
     $args = "$xunitDir"
+    $args += " -log:$logFilePath"
```
Comment (jasonmalinowski) [inline build/scripts/build.ps1]: quoting for spaces in the path?

File: build/scripts/edit-designers.ps1 [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+#
+# This script helps with the times where XAML changes are needed. It will temporarily revert
+# the projects which have XAML files to the legacy project system. The XAML edits can then
+# be processed and the coversion can be reverted.
```
Comment (jasonmalinowski) [inline build/scripts/edit-designers.ps1]: Spelling: "conversion"

File: src/Tools/BuildBoss/ProjectUtil.cs [hunk 14]
```diff
@@ -12,22 +12,39 @@ namespace BuildBoss
...
-        private readonly ProjectKey _key;
-        private readonly XDocument _document;
-        private readonly XmlNamespaceManager _manager;
-        private readonly XNamespace _namespace;
+        internal ProjectKey Key { get; }
+        internal XDocument Document { get; }
+        internal XmlNamespaceManager Manager { get; }
+        internal XNamespace Namespace { get; }
+
+        public bool IsNewSdk => Document.XPathSelectElements("//mb:TargetFramework", Manager).FirstOrDefault() != null;
```
Comment (jasonmalinowski) [inline src/Tools/BuildBoss/ProjectUtil.cs]: .Any()?

File: build/Targets/Imports.targets (hunk 9 above)
Comment (srivatsn) [inline build/Targets/Imports.targets]: You should remove this - this will mean that the dependencies node in solution explorer won't function.

**Comment by sharwell** [comment]
@jaredpar **Imports.targets** needs to be updated to add the `UnitTestContainer` capability to the other unit test projects which were previously not given this capability.

File: build/Targets/Imports.targets (hunk 3 above)
Comment (jaredpar) [inline build/Targets/Imports.targets]: I'll add one in the cleanup change. 

https://github.com/dotnet/roslyn/issues/20526

File: Compilers.sln (hunk 8 above)
Comment (jaredpar) [inline Compilers.sln]: Yes. This change also added BuildBoss enforcement of the GUIDs and it showed a lot of items we missed.

File: build/Targets/Imports.targets (hunk 9 above)
Comment (jaredpar) [inline build/Targets/Imports.targets]: I will test this out later today and if Roslyn performs fine will remove the item.

File: build/Targets/Imports.targets (hunk 10 above)
Comment (jaredpar) [inline build/Targets/Imports.targets]: The commit which added this contains all of the details on why

https://github.com/dotnet/roslyn/pull/20333/commits/4d69e674335f463210cb3054b03e5b2b0967b91d

This is true for virtually every change here. Tried to keep them small and well documented to the reasons.

File: build/Targets/Settings.props (hunk 11 above)
Comment (jaredpar) [inline build/Targets/Settings.props]: Here is the history of this change:

1. The old SDK disabled it everywhere.
1. The new SDK did not and I had to globally override it.
1. A random version of VS broke my override in 2 and a developer incorrectly checked in a whole bunch of suppressions
1. I fixed the global suppression and undid 3. 

So basically this is the old way of doing things 😦

**Comment by jaredpar** [comment]
@sharwell can you give a full sample of what the element should look like? Not familiar with capabilities.

**Comment by sharwell** [comment]
@jaredpar There's one already in Imports.targets. Copy that one up to the previous case.
